### PR TITLE
Replaced node:http/https on undici/fetch (node native)

### PR DESCRIPTION
**Pull request by Bespaliy**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md
- [ ] update .d.ts typings

### CODE REVIEW

**Comment by Tuarisa** [comment]
For node versions 18 and higher fetch is native, no need to import undici just for fetch for all users. 
May be you can add undici as an optional dependencies and check, if you need import undci or use native? Just an idea.

File: lib/network.js [hunk 1]
```diff
@@ -37,31 +36,25 @@ const intToIp = (int) => {
...
-      }, reject);
+      try {
+        return await res.json();
+      } catch (error) {
+        throw new Error(error);
+      }
+    })
+    .catch((error) => {
+      throw new Error(error);
     });
```
Comment (tshemsedinov) [inline lib/network.js:57]:
```suggestion
    );
```

File: package-lock.json [hunk 2]
```diff
@@ -92,6 +93,15 @@
...
     },
+    "node_modules/@fastify/busboy": {
+      "version": "2.1.0",
+      "resolved": "https://registry.npmjs.org/@fastify/busboy/-/busboy-2.1.0.tgz",
+      "integrity": "sha512-+KpH+QxZU7O4675t3mnkQKcZZg56u+K/Ct2K+N2AZYNVK8kyeo/bI18tI8aPm3tvNNRyTWfj6s5tnGNlcbQRsA==",
+      "dev": true,
+      "engines": {
+        "node": ">=14"
+      }
+    },
```
Comment (tshemsedinov) [inline package-lock.json:104]:
```suggestion
```

File: package-lock.json [hunk 3]
```diff
@@ -3135,6 +3145,18 @@
...
+      "resolved": "https://registry.npmjs.org/undici/-/undici-6.0.1.tgz",
+      "integrity": "sha512-eZFYQLeS9BiXpsU0cuFhCwfeda2MnC48EVmmOz/eCjsTgmyTdaHdVsPSC/kwC2GtW2e0uH0HIPbadf3/bRWSxw==",
+      "dev": true,
+      "dependencies": {
+        "@fastify/busboy": "^2.0.0"
+      },
+      "engines": {
+        "node": ">=18.0"
+      }
+    },
```
Comment (tshemsedinov) [inline package-lock.json:3159]:
```suggestion
```

File: lib/network.js [hunk 4]
```diff
@@ -37,31 +35,19 @@ const intToIp = (int) => {
...
-        }
-      }, reject);
-    });
-    req.on('error', reject);
-    if (body) req.write(body);
-    req.end();
-  });
+      return await res.json();
+    })
+    );
```
Comment (tshemsedinov) [inline lib/network.js:50]:
```suggestion
    });
```

**Comment by Bespaliy** [comment]
Is following implantation suitable?
```
const httpApiCall = (url, { method = 'POST', body }) => {
  const mimeType = 'application/json';
  const len = body ? Buffer.byteLength(body) : 0;
  const headers = { 'Content-Type': mimeType, 'Content-Length': len };
  return fetch(url, { method, headers, body }).then((res) => {
    const code = res.status;
    if (code !== 200) {
      const dest = `for ${method} ${url}`;
      throw new Error(`HTTP status code ${code} ${dest}`);
    }
    return res.json();
  });
};
```

**Comment by tshemsedinov** [comment]
> Is following implantation suitable?

We need return await fetch for consistent stack traces.
Now landing.

This will fix code style https://github.com/metarhia/metautil/pull/226
because of eslint rule `no-return-await` off

**Comment by tshemsedinov** [comment]
@Bespaliy see fix: https://github.com/metarhia/metautil/pull/227

**Pull request merged**